File: LandmarkBookwithRW/app/src/main/java/com/ozgortech/landmarkbookwithrw/MainActivity.java
package com.ozgortech.landmarkbookwithrw;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.view.View;
import android.widget.ListView;

import com.ozgortech.landmarkbookwithrw.databinding.ActivityMainBinding;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    //Bu sefer landmark book uygulamamızı listview ile değil de recycler view ile yapacağız
    //Maalesef RecyclerView kendi sınıfından miras alan bir adaptör yapmamızı istiyor
    //İşin zor kısmı bu işte. Kendimiz adaptör için bir sınıf oluşturacağız
    private ActivityMainBinding binding;

    ArrayList<Landmark> landmarks;
    ArrayList<String> landmarkNames;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);

        landmarks = new ArrayList<>();


        Landmark pisa = new Landmark("Pisa","Italy",R.drawable.pisa);
        Landmark eiffel = new Landmark("Eiffel","France",R.drawable.eiffel);
        Landmark kizKulesi = new Landmark("Kız Kulesi","Turkey",R.drawable.kizkulesi);
        Landmark londonEye = new Landmark("London Eye","UK",R.drawable.londoneye);

        landmarks.add(pisa);
        landmarks.add(eiffel);
        landmarks.add(kizKulesi);
        landmarks.add(londonEye);


        binding.landmarkRecycle.setLayoutManager(new LinearLayoutManager(this));//recycler
        //view bir grid şeklinde birde linear layout şeklinde olabiliyor. Biz alt alta metin yazdıra
        //cağımızdan layoutunu linear olarak ayarladık.
        LandmarkAdapter adapter = new LandmarkAdapter(landmarks);//adaptörümüzü oluşturduk
        binding.landmarkRecycle.setAdapter(adapter);//adaptörünü set ettik


    }
}

File: LandmarkBookKotlin/app/src/main/java/com/ozgortech/landmarkbookkotlin/MainActivity.kt
package com.ozgortech.landmarkbookkotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import com.ozgortech.landmarkbookkotlin.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    lateinit var binding : ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        var landmarkList = ArrayList<Landmark>()

        var eiffel = Landmark("Eiffel","France",R.drawable.eiffel)
        var pisa = Landmark("Pisa","Italy",R.drawable.pisa)
        var kizKulesi = Landmark("Kız kulesi","Turkey",R.drawable.kizkulesi)
        var londonEye = Landmark("London Eye","England",R.drawable.londoneye)

        landmarkList.add(eiffel)
        landmarkList.add(pisa)
        landmarkList.add(kizKulesi)
        landmarkList.add(londonEye)

        binding.recyclerView.layoutManager = LinearLayoutManager(this)

        var adapter = LandmarkAdapter(landmarkList)
        binding.recyclerView.adapter = adapter

        // Singleton

        MySingleton.myLandmark = null

        // Kotlinde singleton kullanmak javaya göre çok daha kolay
        // dosyalardan bir tane obje oluşturuyoruz onun içine yazılan her değişken
        // her yerde erişilebilir oluyor (gerçi riskli gibi ama neyse hatta biraz farklı yani)
    }
}